testing/integration-tests/src/full_client/storage/mod.rs

```diff
@@ -125,3 +125,33 @@ async fn storage_n_map_storage_lookup() -> Result<(), subxt::Error> {
     assert_eq!(entry.map(|a| a.amount), Some(123));
     Ok(())
 }
+
+#[tokio::test]
+async fn storage_runtime_wasm_code() -> Result<(), subxt::Error> {
+    let ctx = test_context().await;
+    let api = ctx.client();
+    let wasm_blob = api.storage().at_latest().await?.runtime_wasm_code().await?;
+    assert!(wasm_blob.len() > 1000); // the wasm should be super big
+    Ok(())
+}
+
+#[tokio::test]
+async fn storage_pallet_storage_version() -> Result<(), subxt::Error> {
+    let ctx = test_context().await;
+    let api = ctx.client();
+
+    // cannot assume anything about version number, but should work to fetch it
+    let _version = api
+        .storage()
+        .at_latest()
+        .await?
+        .storage_version("System")
```

Comment:
Crazy thought here; could we use this to do a quick version validation? Similar to our hashing, I guess that would require substrate to constantly modify these numbers appropriately with changes.

Reply:
When would that version validation happen? I guess we would need to make about 30 concurrent calls to validate all pallets. Or we might just validate the pallet whenever a storage query/call is made for a specific pallet?

Reply:
We already validate the keys and value types on a per-entry basis so probably what we have is already better (I imagine the version might increment for every storage change, but not really sure what it represents exactly?)